api_blogs/lib/api_blogs_web/controllers/post_controller.ex

```diff
@@ -33,8 +38,15 @@ defmodule ApiBlogsWeb.PostController do
   end
 
   def show(conn, %{"id" => id}) do
-    post = Blog.get_post!(id)
-    render(conn, "show.json", post: post)
+    try do
+      post_user =
+        id
+        |> Blog.get_post!()
```

Comment:
Aqui está usando a função `Blog.get_post!(id)` que na usa implementação usa `Repo.get!()`. 

Em elixir é comum que as funções normalmente não levantem exceções quando falham e sim retornem uma tupla com `{:error, reason}`. Em contrapartida costuma-se implementar uma outra versão da mesma função com o caractere "!" na frente do nome indicando que ela pode levantar exceções.

A função `Repo.get()` funciona nesse padrão tendo sua contraparte `Repo.get!()`. Isso é interessante pois permite a nós consumidores dessa `lib` decidir o que deve ser tratado como uma exceção ou um erro. Uma exceção é um comportamento realmente inesperado no sistema e um erro é um problema menos sério, algo que pode inclusive estar modelado no domínio do sistema.

Caso um `post` não seja encontrado talvez seja mais simples usar erros e não exceções, isso evita usar defensivas como `try` e `rescue`.

alguns links interessantes sobre:
- [elixir-school](https://elixirschool.com/pt/lessons/intermediate/error_handling)
- [elixir-lang-guide](https://elixir-lang.org/getting-started/try-catch-and-rescue.html)

Introdução do apêndice de _Exceptions_ do livro _[Programming Elixir 1.6](https://www.amazon.com.br/Programming-Elixir-1-6-Dave-Thomas/dp/1680502999)_:

> Exceptions: raise and try, catch and throw
> Elixir (like Erlang) takes the view that errors should normally be fatal to the processes in which they occur. A typical Elixir application’s design involves many processes, which means the effects of an error will be localized. A supervisor will detect the failing process, and the restart will be handled at that level.
> For that reason, you won’t find much exception-handling code in Elixir pro- grams. Exceptions are raised, but you rarely catch them.
> Use exceptions for things that are exceptional—things that should never happen.
> Exceptions do exist. This appendix is an overview of how to generate them and how to catch them when they occur.